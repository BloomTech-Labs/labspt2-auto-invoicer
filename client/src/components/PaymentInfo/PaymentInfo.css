.form {
  width: 300px;
}

.form fieldset {
  border: 1px solid black;
  border-radius: 5px;
  box-shadow: 3px 3px 3px gray;
  display: flex;
  flex-direction: column;
}

.form fieldset input {
  border: 1px solid inherit;
  border-radius: 5px;
  margin: 10px auto;
  font-size: 1.2rem;
}
