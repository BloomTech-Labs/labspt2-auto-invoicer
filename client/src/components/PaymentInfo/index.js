import React from "react";

// import css here
import "./PaymentInfo.css";

const PaymentInfo = props => {
  const { cc, exp, cvv, title } = props.paymentInfo;
  const { changeHandler } = props;

  return (
    <fieldset className="payment_info">
      <legend>{title}</legend>
      <input
        type="text"
        onChange={changeHandler}
        name="cc"
        value={cc}
        placeholder="Credit Card #"
        maxLength={16}
        pattern="[0-9]{16}"
        required
      />
      <input
        type="text"
        onChange={changeHandler}
        name="exp"
        value={exp}
        placeholder="Expiration Date"
        maxLength={4}
        pattern="[0-9]{4}"
        required
      />
      <input
        type="text"
        onChange={changeHandler}
        name="cvv"
        value={cvv}
        placeholder="Cvv"
        pattern="[0-9]{3}"
        maxLength={3}
        required
      />
    </fieldset>
  );
};

export default PaymentInfo;
