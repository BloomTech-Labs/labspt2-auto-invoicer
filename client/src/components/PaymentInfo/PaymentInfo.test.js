import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import PaymentInfo from "../PaymentInfo";

describe("<PaymentInfo />", () => {
  it("renders Payment Info component", () => {
    const { asFragment } = render(
      <PaymentInfo paymentInfo={{ cc: "", exp: "", cvv: "", title: "" }} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
