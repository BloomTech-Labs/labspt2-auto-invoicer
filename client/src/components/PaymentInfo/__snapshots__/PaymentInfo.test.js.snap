// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentInfo /> renders Payment Info component 1`] = `
<DocumentFragment>
  <fieldset
    class="payment_info"
  >
    <legend />
    <input
      maxlength="16"
      name="cc"
      pattern="[0-9]{16}"
      placeholder="Credit Card #"
      required=""
      type="text"
      value=""
    />
    <input
      maxlength="4"
      name="exp"
      pattern="[0-9]{4}"
      placeholder="Expiration Date"
      required=""
      type="text"
      value=""
    />
    <input
      maxlength="3"
      name="cvv"
      pattern="[0-9]{3}"
      placeholder="Cvv"
      required=""
      type="text"
      value=""
    />
  </fieldset>
</DocumentFragment>
`;
